core/src/main/java/com/linecorp/armeria/internal/http/Http1ObjectEncoder.java

```diff
@@ -344,9 +343,23 @@ private ChannelFuture doWriteSplitData(
         }
 
         ctx.flush();
+
+        if (data instanceof ByteBufHttpData) {
+            ((ByteBufHttpData) data).buf().release();
```

Comment:
You could release in `dataChunk()` because you used `retainedSlice()` there, no?